test/parallel/test-parse-args.mjs

```diff
@@ -211,73 +211,42 @@ test('order of option and positional does not matter (per README)', () => {
-
-  const expected = { values: { __proto__: null, foo: true },
-                     positionals: [] };
-  assert.deepStrictEqual(result, expected);
-  process.argv = holdArgv;
-  process.execArgv = holdExecArgv;
-});
-
-test('correct default args when use node -e', () => {
-  const holdArgv = process.argv;
-  process.argv = [process.argv0, '--foo'];
-  const holdExecArgv = process.execArgv;
-  process.execArgv = ['-e', '0'];
-  const result = parseArgs({ strict: false });
-
-  const expected = { values: { __proto__: null, foo: true },
-                     positionals: [] };
-  assert.deepStrictEqual(result, expected);
-  process.argv = holdArgv;
-  process.execArgv = holdExecArgv;
-});
-
-test('correct default args when use node --eval', () => {
-  const holdArgv = process.argv;
-  process.argv = [process.argv0, '--foo'];
-  const holdExecArgv = process.execArgv;
-  process.execArgv = ['--eval', '0'];
-  const result = parseArgs({ strict: false });
-  const expected = { values: { __proto__: null, foo: true },
-                     positionals: [] };
-  assert.deepStrictEqual(result, expected);
-  process.argv = holdArgv;
-  process.execArgv = holdExecArgv;
-});
-
-test('correct default args when normal arguments', () => {
-  const holdArgv = process.argv;
-  process.argv = [process.argv0, 'script.js', '--foo'];
-  const holdExecArgv = process.execArgv;
-  process.execArgv = [];
-  const result = parseArgs({ strict: false });
+suite('correct default args', () => {
+  suite('with CLI flags', () => {
+    const evalCode = "JSON.stringify(require('util').parseArgs({ strict: false }).values)";
+    const evalCodePrinted = `process.stdout.write(${evalCode})`;
+    const execArgsTests = {
+      '-e <script>': ['-e', evalCodePrinted],
+      '-p <script>': ['-p', evalCode],
+      '-pe <script>': ['-pe', evalCode],
+      '--eval <script>': ['--eval', evalCodePrinted],
+      '--eval=<script>': [`--eval=${evalCodePrinted}`],
+      '--print <script>': ['--print', evalCode],
+      '--print --eval <script>': ['--print', '--eval', evalCode],
+      '--print --eval=<script>': ['--print', `--eval=${evalCode}`],
+    };
+    for (const description in execArgsTests) {
+      const execArgs = execArgsTests[description];
+      test(description, async () => {
+        const { code, signal, stderr, stdout } = await spawnPromisified(
+          process.execPath,
+          [...execArgs, '--', '--foo', '--bar']);
+        assert.strictEqual(code, 0);
+        assert.strictEqual(signal, null);
+        assert.strictEqual(stderr, '');
+        assert.deepStrictEqual(JSON.parse(stdout), { foo: true, bar: true });
+      });
+    }
+  });
 
-  const expected = { values: { __proto__: null, foo: true },
-                     positionals: [] };
-  assert.deepStrictEqual(result, expected);
-  process.argv = holdArgv;
-  process.execArgv = holdExecArgv;
+  test('without CLI flags', () => {
+    assert.strictEqual(process.execArgv.length, 0);
```

Comment:
My fault entirely, I tested this commit outside of the CI runner.

This assertion isn't necessary, it's probably easiest just to remove it. The PR should then be green.


```suggestion
```